internal/elfexec/elfexec.go

```diff
@@ -189,9 +188,21 @@ func GetBase(fh *elf.FileHeader, loadSegment *elf.ProgHeader, stextOffset *uint6
 	switch fh.Type {
 	case elf.ET_EXEC:
 		if loadSegment == nil {
-			// Fixed-address executable, no adjustment.
+			// Assume fixed-address executable and so no adjustment.
 			return 0, nil
 		}
+		if stextOffset == nil && start > 0 && start < 0x8000000000000000 {
```

Comment:
Thanks for the explanation on the meaning of stextOffset.
If that value is optional, do you think it is safe to handle all user mode addresses here, regardless if stextOffset is nil or not?
If stextOffset != nil, then we fall into one of the crazy special cases below.

Reply:
Oh, rereading your comment I see that the _stext symbol is in the vmlinux image.